Makefile

```diff
@@ -1,44 +1,72 @@
 # Package configuration
 PROJECT = code-annotation
 COMMANDS = server/cmd/code-annotation
+DEPENDENCIES = github.com/golang/dep/cmd/dep github.com/jteeuwen/go-bindata
 
 # Including ci Makefile
-MAKEFILE = Makefile.main
 CI_REPOSITORY = https://github.com/src-d/ci.git
-CI_FOLDER = .ci
+CI_FOLDER = $(shell pwd)/.ci
+MAKEFILE := $(CI_FOLDER)/Makefile.main
 
 # Tools
 YARN = yarn
-REMOVE = rm -rf
+GODEP = dep
+GOLINT = golint
+GOVET = go vet
+BINDATA = go-bindata
 
 SERVER_URL ?= /api
 API_PORT ?= 8080
 
 $(MAKEFILE):
-	@git clone --quiet $(CI_REPOSITORY) $(CI_FOLDER); \
-	cp $(CI_FOLDER)/$(MAKEFILE) .;
+	@git clone --quiet --depth 1 -b v1 $(CI_REPOSITORY) $(CI_FOLDER);
 
 -include $(MAKEFILE)
 
 # set enviroment variables from .env file
 include .env
 export $(shell sed 's/=.*//' .env)
 
-dependencies-frontend: dependencies
+# Frontend
+dependencies-frontend:
 	$(YARN)	install
 
 test-frontend: dependencies-frontend
 	$(YARN) test
 
-lint: dependencies-frontend
+lint-frontend: dependencies-frontend
 	$(YARN) lint
 
-build: dependencies-frontend
+build-frontend: dependencies-frontend
 	REACT_APP_SERVER_URL=$(SERVER_URL) $(YARN) build
 
-## Compiles the dashboard assets, and serve the dashboard through its API
-serve: build
-	go run server/cmd/code-annotation/*
+dev-frontend: dependencies-frontend
+	$(YARN) start
+
+# Backend
+dependencies-backend: $(DEPENDENCIES)
+	$(GODEP) ensure
+
+build-backend: dependencies-backend
+
+lint-backend: dependencies-backend
+	$(GOLINT) ./server/...
+	$(GOVET) ./server/...
+	
+bindata:
+	$(BINDATA) -o ./server/cmd/code-annotation/asset.go build/static/... build/*.json build/*.png build/index.html
 
+release-build: build-frontend build-backend bindata packages docker-build
```

Comment:
release build should be `packages` something like:

```
packages: build-frontend  dependencies-backend bindata packages docker-build
```

Reply:
Also, you need to force the order of the dependencies with `rule: | one two three` otherwise if someone use `-j n` will fail